### DISCUSSION TITLE
Can we load models downloaded directly from tensor flow hub?

### DISCUSSION THREAD

**Discussion by pulkitmalhotra1995** [JohnSnowLabs/spark-nlp#2936] [General]
I am new to spark nlp family, trying to make sense of things. I see spark NLP by default downloads https://tfhub.dev/google/universal-sentence-encoder/2 USE embeddings, but these are old, I downloaded the latest from https://tfhub.dev/google/universal-sentence-encoder/4, is there any way I can load these directly using spark NLP? I tried using .load but it asks for metadata directory to present.

**Comment by maziyarpanahi** [comment]
Load function is to load Spark NLP models not TensorFlow SavedModel. For that we use .loadSavedModel. 

However, we do the export/import ourselves do to TF v1 conversions.

In the new release this can be possible by users but right now the process is long and tedious.

**Reply by pulkitmalhotra1995** [reply]
Thanks @maziyarpanahi .loadSavedModel does the job. But one observation from it, it does load bert models downloaded from tf hub, but fails to load  USE v4 (https://tfhub.dev/google/universal-sentence-encoder/4).
logs: IllegalArgumentException: No Operation named [init_all_tables] in the Graph

**Reply by maziyarpanahi** [reply]
As I said, currently is not possible for you to do it, it has to be done by us. Also, that model is TF v2 SavedModel, which is not compatible with the current release and will be supported in the next release(s). 

We are working on making this easy and possible and started with popular annotators like BERT, RoBERTa, DistilBERT, etc. We will get to UniversalSentenceEncoder in a few weeks so you can just download directly from TF Hub and just use .loadSavedModel yourself in 1 line without needing to set or modify anything behind the scene in TF v2.

**Reply by pulkitmalhotra1995** [reply]
Thanks a lot

**Reply by pulkitmalhotra1995** [reply]
@maziyarpanahi is this possible now with the new release?

**Reply by maziyarpanahi** [reply]
@pulkitmalhotra1995 If the model you need is from TF Hub, it is still under development. The models there are not unified, each has its own inputs/ output(s) / then many other stuff. We are working on finding a good and easy way like HuggingFace.

We will update this guide by changing the table and adding notebooks once it's possible: 
[Import Transformers into Spark NLP](https://github.com/JohnSnowLabs/spark-nlp/discussions/5669)
